controllers/toppingsController.js

```diff
@@ -0,0 +1,104 @@
+
+const { promisify } = require("util");
+
+const Toppings = require("../models/toppingsModels");
+const catchAsync = require("../utils/catchAsync");
+const AppError = require("../utils/appError");
+
+//list toppings 
+exports.listToppings = catchAsync(async (req, res, next) => {
+  const name = req.query.name;
+  var condition = name ? { name: { $regex: new RegExp(name), $options: "i" } } : {};
+
+  // find the toppings by ID
+  Toppings.find(condition)
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      return next(
+        new AppError("Error occurred while retrieving toppings.", 500)
+      );
+    });
+});
+
+//create toppings
+exports.createToppings = catchAsync(async (req, res, next) => {
+  // Validate request
+  if (!req.body.name) {
+      return next(
+      new AppError("Request can not be empty!", 400)
+    );
+  }
+
+  // Create a Topping
+  const toppings = new Toppings({
+    name: req.body.name,
+    price: req.body.price
+  });
+
+  // Save topping in the database
+  toppings
+    .save(toppings)
```

Comment:
We should use toppings.create() instead of .save()


Refer to this snippet
  const newProduct = await Product.create(req.body);

  res.status(201).json({
    status: 'success',
    data: {
      products: newProduct,
    },
  });

Reply:
Fixed